### PR TITLE
Extract RSIZ marker on decoding

### PR DESCRIPTION
**Pull request by remia**
This PR address the issue described in #1198.

### CODE REVIEW

File: src/lib/openjp2/openjpeg.h [hunk 1]
```diff
@@ -667,6 +667,8 @@ typedef struct opj_image_comp {
  * Defines image data and characteristics
  * */
 typedef struct opj_image {
+    /** RSIZ: capabilities */
+    OPJ_UINT16 rsiz;
```
Comment (rouault) [inline src/lib/openjp2/openjpeg.h]: to avoid ABI issues this should be moved as the last component of the structure

Comment (remia) [reply src/lib/openjp2/openjpeg.h]: Oh right thanks, did not trigger an issue in my CI build weird.

**Comment by remia** [comment]
The CI is bound to fail when making update like this (impact on the ABI), right ?

**Comment by rouault** [comment]
I've solved the unrelated abi-checker issue that plagued all builds recently, and re-ran your build. So now we have a diagnostic related to your change
```
<td>Field <b>rsiz</b> has been added to this type.</td>

<td>This field will not be initialized by old clients.<br/><b>NOTE</b>: this field should be accessed only from the new library functions, otherwise it may result in crash or incorrect behavior of applications.</td>
```
It is flagged as low risk problem, and indeed as all public functions in the API manipulates opj_image_t* as a pointer, and it is unexpected that user code would allocate it with sizeof(opj_image_t), this should be safe.
@detonin How are we supposed to handle such cases ?

**Comment by remia** [comment]
Thanks @rouault, just let me know if there is anything to add.

**Comment by ValZapod** [comment]
So lets merge that?

**Comment by remia** [comment]
Thanks for linking that commit @ValZapod, do you append to have context about this ?

According to my tests, it doesn't work on MXF wrapped JP2K Cinema files, eg:

`ffmpeg -v debug -vcodec libopenjpeg -i my_j2k.mxf`

This is because of wrong RSIZ key and missing JP2KEssenceSubDescriptor lookup group. Here is a patch I made to fix that, can be applied to current master branch:

```
diff --git a/libavformat/mxfdec.c b/libavformat/mxfdec.c
index 6f6e8d586c..cd85fbedff 100644
--- a/libavformat/mxfdec.c
+++ b/libavformat/mxfdec.c
@@ -325,7 +325,7 @@ static const uint8_t mxf_encrypted_essence_container[]     = { 0x06,0x0e,0x2b,0x
 static const uint8_t mxf_random_index_pack_key[]           = { 0x06,0x0e,0x2b,0x34,0x02,0x05,0x01,0x01,0x0d,0x01,0x02,0x01,0x01,0x11,0x01,0x00 };
 static const uint8_t mxf_sony_mpeg4_extradata[]            = { 0x06,0x0e,0x2b,0x34,0x04,0x01,0x01,0x01,0x0e,0x06,0x06,0x02,0x02,0x01,0x00,0x00 };
 static const uint8_t mxf_avid_project_name[]               = { 0xa5,0xfb,0x7b,0x25,0xf6,0x15,0x94,0xb9,0x62,0xfc,0x37,0x17,0x49,0x2d,0x42,0xbf };
-static const uint8_t mxf_jp2k_rsiz[]                       = { 0x06,0x0e,0x2b,0x34,0x02,0x05,0x01,0x01,0x0d,0x01,0x02,0x01,0x01,0x02,0x01,0x00 };
+static const uint8_t mxf_jp2k_rsiz[]                       = { 0x06,0x0e,0x2b,0x34,0x01,0x01,0x01,0x0a,0x04,0x01,0x06,0x03,0x01,0x00,0x00,0x00 };
 static const uint8_t mxf_indirect_value_utf16le[]          = { 0x4c,0x00,0x02,0x10,0x01,0x00,0x00,0x00,0x00,0x06,0x0e,0x2b,0x34,0x01,0x04,0x01,0x01 };
 static const uint8_t mxf_indirect_value_utf16be[]          = { 0x42,0x01,0x10,0x02,0x00,0x00,0x00,0x00,0x00,0x06,0x0e,0x2b,0x34,0x01,0x04,0x01,0x01 };
 
@@ -2746,6 +2746,7 @@ static const MXFMetadataReadTableEntry mxf_metadata_read_table[] = {
     { { 0x06,0x0e,0x2b,0x34,0x02,0x53,0x01,0x01,0x0d,0x01,0x01,0x01,0x01,0x01,0x42,0x00 }, mxf_read_generic_descriptor, sizeof(MXFDescriptor), Descriptor }, /* Generic Sound */
     { { 0x06,0x0e,0x2b,0x34,0x02,0x53,0x01,0x01,0x0d,0x01,0x01,0x01,0x01,0x01,0x28,0x00 }, mxf_read_generic_descriptor, sizeof(MXFDescriptor), Descriptor }, /* CDCI */
     { { 0x06,0x0e,0x2b,0x34,0x02,0x53,0x01,0x01,0x0d,0x01,0x01,0x01,0x01,0x01,0x29,0x00 }, mxf_read_generic_descriptor, sizeof(MXFDescriptor), Descriptor }, /* RGBA */
+    { { 0x06,0x0e,0x2b,0x34,0x02,0x53,0x01,0x01,0x0d,0x01,0x01,0x01,0x01,0x01,0x5a,0x00 }, mxf_read_generic_descriptor, sizeof(MXFDescriptor), Descriptor }, /* JPEG2000PictureSubDescriptor */
     { { 0x06,0x0e,0x2b,0x34,0x02,0x53,0x01,0x01,0x0d,0x01,0x01,0x01,0x01,0x01,0x48,0x00 }, mxf_read_generic_descriptor, sizeof(MXFDescriptor), Descriptor }, /* Wave */
     { { 0x06,0x0e,0x2b,0x34,0x02,0x53,0x01,0x01,0x0d,0x01,0x01,0x01,0x01,0x01,0x47,0x00 }, mxf_read_generic_descriptor, sizeof(MXFDescriptor), Descriptor }, /* AES3 */
     { { 0x06,0x0e,0x2b,0x34,0x02,0x53,0x01,0x01,0x0d,0x01,0x01,0x01,0x01,0x01,0x51,0x00 }, mxf_read_generic_descriptor, sizeof(MXFDescriptor), Descriptor }, /* MPEG2VideoDescriptor */
```

Then, even with that patch, the pixel format detected by the demuxer is not communicated to the decoder, unless I'm missing something, so that it doesn't really matter. I'm not an ffmpeg expert so I didn't found how this kind of communication are supposed to occur.

FYI, I currently already used the patch in this pull-request with this correspond FFmpeg patch with success (https://github.com/remia/ffmpeg-docker/blob/master/0001-Interpret-cinema-profiles-as-XYZ.patch), granted that there is probably a cleaner of doing this.

May something worth posting on ffmpeg-devel ?

**Comment by ValZapod** [comment]
> posting on ffmpeg-devel

Yes!! See, some guys are righ now trying to do the native decoder in ffmpeg. https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=&submitter=&state=&q=JPEG2000&archive=&delegate=

**Comment by remia** [comment]
For reference, the patch I proposed to fix the pixel format identification when using libopenjpeg decoder in ffmpeg in here http://ffmpeg.org/pipermail/ffmpeg-devel/2020-September/269088.html It doesn't require any update of libopenjpeg itself by moving the RSIZ extraction to the demuxer (mxf).

**Comment by ValZapod** [comment]
You patch failed to apply https://patchwork.ffmpeg.org/project/ffmpeg/patch/CAH6sF9_T2T5Em4MRrQYk=S+7QWvMPAWnSSYnxEf4mcFeB3rFLw@mail.gmail.com/

**Comment by ValZapod** [comment]
Is that "an XAVC file CDCI descriptor" in ffmpeg FATE?

https://patchwork.ffmpeg.org/project/ffmpeg/patch/20200913224218.60972-1-remiachard@gmail.com/

And about the pull request you need to close and reopen it to repeat the Travis CI.

**Comment by remia** [comment]
The file is mxf/Sony-00001.mxf IIRC, it's not something specific as the layout follow the XAVC specs and other XAVC files from Sony professional cameras are similar.

I did stop thinking about this due to the lack of interest, not planning on pursuing this for now unless there are new inputs / suggestion / help.

**Comment by ValZapod** [comment]
> there are new inputs

I think it is already good. Just ping the devs in ffmpeg. And this one is NOT actually breacking ABI.

**Comment by ValZapod** [comment]
@remia FYI. I patched ffmpeg's mxf_jp2k_rsiz https://github.com/FFmpeg/FFmpeg/commit/91ba771a3fee9757273416789ba01e1a7c62890b

Now, I suppose that may be not enough, but whatever. Will test and update this comment.

**Comment by remia** [comment]
Thanks @ValZapod, yeah I guess it'll not be enough but that's a start.

The patch I made that don't rely on openjpeg to merge this PR was https://patchwork.ffmpeg.org/project/ffmpeg/patch/20200913224218.60972-1-remiachard@gmail.com/ and include a few other stuff, IIRC there was some issue related to the way I updated `mxf_read_strong_ref_array` but I'm not such an expert in MXF to know how to solve that easily.

**Pull request closed without merging**